ffi/Gosu_Image.cpp

```diff
@@ -1,168 +1,168 @@
+    std::unique_ptr<Gosu::ImageData> image_data =
+            image->image.data().subimage(left, top, width, height);
+
+    if (!image_data) {
+        return nullptr;
+    }
 
-    return reinterpret_cast<Gosu_Image *>(image_data.get() ? new Gosu::Image(std::move(image_data)) : nullptr);
+    return new Gosu_Image{Gosu::Image{std::move(image_data)}};
 }
 
-void Gosu_Image_create_from_tiles(const char *source, int tile_width, int tile_height, void function(void *data, Gosu_Image *image), void *data, unsigned image_flags)
+GOSU_FFI_API void Gosu_Image_create_from_tiles(const char* source, int tile_width, int tile_height,
+                                               void function(void*, Gosu_Image*), void* data,
+                                               unsigned image_flags)
 {
-    std::vector<Gosu::Image> gosu_images = Gosu::load_tiles(source, tile_width, tile_height, image_flags);
+    std::vector<Gosu::Image> gosu_images =
+            Gosu::load_tiles(source, tile_width, tile_height, image_flags);
 
-    for (Gosu::Image &img : gosu_images) {
-        function(data, reinterpret_cast<Gosu_Image *>(new Gosu::Image(img)));
+    for (Gosu::Image& img : gosu_images) {
+        function(data, new Gosu_Image{img});
     }
 }
 
-void Gosu_Image_create_tiles_from_image(Gosu_Image *image, int tile_width, int tile_height, void function(void *data, Gosu_Image *image), void *data, unsigned image_flags)
+GOSU_FFI_API void Gosu_Image_create_tiles_from_image(Gosu_Image* image, int tile_width,
+                                                     int tile_height,
+                                                     void function(void*, Gosu_Image*), void* data,
+                                                     unsigned image_flags)
 {
-    std::vector<Gosu::Image> gosu_images = Gosu::load_tiles(reinterpret_cast<Gosu::Image *>(image)->data().to_bitmap(), tile_width, tile_height, image_flags);
+    std::vector<Gosu::Image> gosu_images =
+            Gosu::load_tiles(image->image.data().to_bitmap(), tile_width, tile_height, image_flags);
 
-    for (Gosu::Image &img : gosu_images) {
-        function(data, reinterpret_cast<Gosu_Image *>(new Gosu::Image(img)));
+    for (Gosu::Image& img : gosu_images) {
+        function(data, new Gosu_Image{img});
     }
 }
 
+GOSU_FFI_API void Gosu_Image_destroy(Gosu_Image* image)
+{
+    delete image;
+}
+
 // Properties
-unsigned Gosu_Image_width(Gosu_Image *image)
+
+GOSU_FFI_API unsigned Gosu_Image_width(Gosu_Image* image)
 {
-    return reinterpret_cast<Gosu::Image *>(image)->width();
+    return image->image.width();
 }
 
-unsigned Gosu_Image_height(Gosu_Image *image)
+GOSU_FFI_API unsigned Gosu_Image_height(Gosu_Image* image)
 {
-    return reinterpret_cast<Gosu::Image *>(image)->height();
+    return image->image.height();
 }
 
-Gosu_GLTexInfo *Gosu_Image_gl_tex_info_create(Gosu_Image *image)
+GOSU_FFI_API Gosu_GLTexInfo* Gosu_Image_gl_tex_info_create(Gosu_Image* image)
```

Comment:
At this point I really regret that C++ Gosu uses namespace at all, imagine how easy things would be if the naming convention was just GosuGLTexInfo, GosuImage, GosuWindow and both languages could share the same definition...